caffe2/CMakeLists.txt

```diff
@@ -1187,7 +1187,7 @@ endif()
 # jit/unpickler.cpp need to be compiled only when USE_DISTRIBUTED is set
 if(USE_DISTRIBUTED)
   # Needed to support the inclusion of c10d/Foo.hpp headers.
-  target_include_directories(torch_cpu PUBLIC ${TORCH_SRC_DIR}/lib)
```

Comment:
is there ever a reason to use `TORCH_SRC_DIR` (cc @malfet)? maybe we should just get rid of it to prevent problems like this in the future

Reply:
I think for other cmake files we basically use `TORCH_SRC_DIR` as an alias for `CMAKE_CURRENT_SOURCE_DIR` but maybe we shouldn't any more.

Also for some reason caffe2 has a different definition of what TORCH_SRC_DIR actually is:
https://github.com/pytorch/pytorch/blob/65f33ec85c2a7d8fb9bf582017d3170bf89e6c12/caffe2/CMakeLists.txt#L325-L330

Compared with the torch cmake:
https://github.com/pytorch/pytorch/blob/65f33ec85c2a7d8fb9bf582017d3170bf89e6c12/torch/CMakeLists.txt#L19-L22

Reply:
Actually I think the mistake here was that the original author didn't include `BUILD_INTERFACE` which should automatically set it to not be hardcoded